### PR TITLE
ENG-2468 Add search functionality to template selection modal in page…

### PR DESCRIPTION
**Pull request by raxkaynan**
… form

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **raxkaynan**

If they are not already assigned, you can assign the PR to them by writing `/assign @raxkaynan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/ui/common/modal/GenericModalContainer.js
Comment (jeffgo10) [inline src/ui/common/modal/GenericModalContainer.js:26]: is there a reason why you wanted to put this extra function prop in `mergeProps`? why not put this line inside `onCloseModal` on dispatchProps?

**Review by jeffgo10 — CHANGES_REQUESTED** [review]
just this 1 concern. I remembered about our rule of code on avoiding `mergeProps`

File: src/ui/common/modal/GenericModalContainer.js
Comment (raxkaynan) [reply src/ui/common/modal/GenericModalContainer.js:26]: `ownProps.onCloseModal()` can't be accessible within `mapDispatchToProps`. That extra function call is added since `FindTemplateModal` now has state and needs to reset state values after the modal has been closed. I don't like `mergeProps` as well but that's the simplest way unless we'd rewrite `GenericModalContainer` and `GenericModal` 😄

**Review by jeffgo10 — APPROVED** [review]
looks good

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1098#pullrequestreview-691609234" title="Approved">jeffgo10</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [jeffgo10]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
